source/meta/kernel-4/metadata.xsd

```diff
@@ -105,6 +106,9 @@
           <xs:complexType>
             <xs:simpleContent>
               <xs:extension base="nonemptycontentStringType">
+                <xs:attribute name="publisherIdentifier" type="xs:string" use="optional" />
+                <xs:attribute name="publisherIdentifierScheme" type="xs:string" use="optional" />
```

Comment:
As far as I know, there isn't a way to enforce this in the XSD itself. This is similar to other cases where *Identifier and *Scheme are implemented as attributes, namely affiliationIdentifier and affiliationIdentifierScheme.

Reply:
Thanks! A few questions:

- Should we take the documentation regarding these properties to be more of an advisement than a requirement? 
- Should we enforce the requirement when validating metadata beyond the XSD? For example, when receiving JSON REST API requests for findable DOIs? In my thinking, the API and the XSD should be aligned as far as metadata validity, but curious if you or others have thoughts on this.

Reply:
If it helps, I was curious and found this in stackoverflow - https://stackoverflow.com/questions/12746049/making-an-attribute-required-only-if-another-attribute-is-set  which basically said it is not possible with xsd 1.0 alone.

Reply:
@codycooperross I think if we could enforce these as requirements, we would, so we've kept the "required" language in the documentation even though it isn't strictly correct (more advisement).

Do we do any validation beyond the XSD for these "if [attribute] is provided, [other attribute] is required" cases currently?

Reply:
> Do we do any validation beyond the XSD for these "if [attribute] is provided, [other attribute] is required" cases currently?

As far as I know, no, aside from perhaps particular restrictions in the Fabrica form. We could conceivably create tighter restrictions in the REST API, but I think this might cause a lot of confusion and technical complexity if users are able to create metadata in XML that they can't in JSON.